akka-actor-typed/src/main/scala/akka/actor/typed/internal/TimerSchedulerImpl.scala

```diff
@@ -29,11 +24,15 @@ import scala.reflect.ClassTag
     scaladsl.Behaviors.setup[T](wrapWithTimers(factory))
   }
 
-  def wrapWithTimers[T](factory: TimerSchedulerImpl[T] ⇒ Behavior[T])(ctx: ActorContext[T]): Behavior[T] = {
-    val timerScheduler = new TimerSchedulerImpl[T](ctx)
-    val behavior = factory(timerScheduler)
-    timerScheduler.intercept(behavior)
-  }
+  def wrapWithTimers[T](factory: TimerSchedulerImpl[T] ⇒ Behavior[T])(ctx: ActorContext[T]): Behavior[T] =
+    ctx match {
+      case ctxImpl: ActorContextImpl[T] ⇒
+        val timerScheduler = ctxImpl.timer
+        val behavior = factory(timerScheduler)
+        timerScheduler.intercept(behavior)
+      case _ ⇒ throw new IllegalArgumentException(s"timers not supported with [${ctx.getClass}]")
```

Comment:
Meaning that it's not supported by `StubbedActorContext`. Might be ok since timing related things shouldn't be tested with sync BehaviorTestKit. Related https://github.com/akka/akka/issues/24686
I guess it wouldn't be difficult to add same in `StubbedActorContext` if needed